Sources/Core/API/Block.swift

```diff
@@ -119,16 +119,35 @@ public final class Block: SceneObject, InstanceHashable, ActionPerformer, ZIndex
         layer.addSublayer(segmentLayers.top)
         layer.addSublayer(segmentLayers.bottom)
 
+        #if os(macOS)
+        segmentLayers.bottom.frame.origin.y = 0
+        #else
         segmentLayers.bottom.frame.origin.y = size.height - segmentLayers.bottom.frame.height
+        #endif
 
         let centerReplicatorLayer = ReplicatorLayer()
+        #if os(macOS)
```

Comment:
All this conditional compilation makes this code really hard to follow IMO (it was already a bit hard to follow, and that's my fault 😅), how about breaking it up into some helper functions and put the conditional compilation inside those functions (will make it a bit more contained)?

Reply:
Absolutely agree with you @JohnSundell, I fell for the old "it's already a bit complicated so what does it matter if I make it a little bit more complicated". My bad! I will take another look and see if I can break up the code a little. It is a little risky since we don't have that many unit tests on Block and it isn't that easy to add either. But at least you get clear visual feedback if it is wrong... 😄

Reply:
Haha yeah, it's a bit tricky to test these purely visual things 🤔  We could always verify the layer hierarchy but I'm not sure if that makes a ton of sense.